server/api/studio/pkg/ws/client.go

```diff
@@ -0,0 +1,285 @@
+	"github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/base"
+	"github.com/vesoft-inc/nebula-studio/server/api/studio/pkg/ecode"
+)
+
+const (
+	// Time allowed to write a message to the peer.
+	writeWait = 10 * time.Second
+
+	// Time allowed to read the next pong message from the peer.
+	pongWait = 60 * time.Second
+
+	// Send pings to peer with this period. Must be less than pongWait.
+	pingPeriod = (pongWait * 9) / 10
+
+	// Maximum message size allowed from peer.
+	maxMessageSize = 16 * 1024
+
+	// send buffer size
+	bufSize = 512
+
+	heartbeatRequest = "1"
+
+	heartbeatResponse = "2"
+)
+
+var (
+	newline = []byte{'\n'}
+	space   = []byte{' '}
+)
+
+var upgrader = websocket.Upgrader{
+	// ReadBufferSize:  1024,
+	// WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+	Error: func(w http.ResponseWriter, r *http.Request, status int, reason error) {
+		w.WriteHeader(status)
+		w.Write([]byte(reason.Error()))
+	},
+}
+
+// Client is a middleman between the websocket connection and the hub.
+type Client struct {
+	hub *Hub
+
+	clientInfo *auth.AuthData
+
+	// The websocket connection.
+	conn *websocket.Conn
+
+	// Buffered channel of outbound messages.
+	send chan []byte
+}
+
+func (c *Client) runNgql(msgReceived *MessageReceive) {
+	msgPost := MessagePost{}
+	msgPost.Header.MsgId = msgReceived.Header.MsgId
+	msgPost.Header.SendTime = time.Now().UnixMilli()
+	msgPost.Body.MsgType = msgReceived.Body.MsgType
+
+	gql, paramList := "", []string{}
+
+	reqGql := msgReceived.Body.Content["gql"]
+	if reqGql != nil {
+		gql, _ = reqGql.(string)
+	}
+
+	reqParamList := msgReceived.Body.Content["paramList"]
+	if reqParamList != nil && reflect.TypeOf(reqParamList).Kind() == reflect.Slice {
+		s := reqParamList.([]interface{})
+		for i := 0; i < len(s); i++ {
+			paramList = append(paramList, s[i].(string))
+		}
+	}
+
+	execute, _, err := dao.Execute(c.clientInfo.NSID, gql, paramList)
+
+	if err != nil {
+		content := map[string]any{
+			"code":    base.Error,
+			"message": err.Error(),
+		}
+		if auth.IsSessionError(err) {
+			content["code"] = ecode.ErrSession.GetCode()
+		}
+		msgPost.Body.Content = &content
+	} else {
+		msgPost.Body.Content = map[string]any{
+			"code":    base.Success,
+			"data":    &execute,
+			"message": "Success",
+		}
+	}
+
+	msgSend, _ := json.Marshal(msgPost)
+	c.send <- msgSend
+}
+
+func (c *Client) runBatchNgql(msgReceived *MessageReceive) {
+	msgPost := MessagePost{}
+	msgPost.Header.MsgId = msgReceived.Header.MsgId
+	msgPost.Header.SendTime = time.Now().UnixMilli()
+	msgPost.Body.MsgType = msgReceived.Body.MsgType
+
+	gqls, paramList := []string{}, []string{}
+	resContentData := make([]map[string]interface{}, 0)
+
+	reqGqls := msgReceived.Body.Content["gqls"]
+	if reqGqls != nil && reflect.TypeOf(reqGqls).Kind() == reflect.Slice {
+		s := reqGqls.([]interface{})
+		for i := 0; i < len(s); i++ {
+			gqls = append(gqls, s[i].(string))
+		}
+	}
```

Comment:
Maybe add some functions for `MessageReceiveBody` is more clearer?
And check the type in those functions.
```
func (b *MessageReceiveBody) GetGqls() (string, error) {
}
func (b *MessageReceiveBody) GetParamList() ([]string, error) {
}
// and so on 
```